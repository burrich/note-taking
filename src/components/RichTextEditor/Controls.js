import React from 'react';
import InlineStyleButton from './InlineStyleButton';
import BlockStyleButton from './BlockStyleButton';

/**
 * Controls component container for StyleButton components.
 * TODO: return an array map
 */
function Controls(props) {
  const inlineStyle       = props.currentStyle.inline;
  const blockStyle        = props.currentStyle.block;
  const toggleInlineStyle = props.toggleStyle.inline;
  const toggleBlockStyle  = props.toggleStyle.block;

  return (
    <div className="RichEditor-controls">
      <InlineStyleButton name="bold"
                         editorStyle={inlineStyle}
                         onToggle={toggleInlineStyle} />

      <InlineStyleButton name="italic"
                         editorStyle={inlineStyle}
                         onToggle={toggleInlineStyle} />

      <InlineStyleButton name="underline"
                         editorStyle={inlineStyle}
                         onToggle={toggleInlineStyle} />

      <InlineStyleButton name="strikethrough"
                         editorStyle={inlineStyle}
                         onToggle={toggleInlineStyle} />

      <BlockStyleButton name="quotes"
                        editorStyle={blockStyle}
                        onToggle={toggleBlockStyle} />

      <BlockStyleButton name="code"
                        editorStyle={blockStyle}
                        onToggle={toggleBlockStyle} />

      <BlockStyleButton name="ul"
                        editorStyle={blockStyle}
                        onToggle={toggleBlockStyle} />

      <BlockStyleButton name="ol"
                        editorStyle={blockStyle}
                        onToggle={toggleBlockStyle} />
    </div>
  );
}

export default Controls;
