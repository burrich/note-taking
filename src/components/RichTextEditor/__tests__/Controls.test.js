import React from 'react';
import { shallow } from 'enzyme';
import Controls from '../Controls';

test('Controls render correctly', () => {
  const currentStyle = {
    inline: new Map(),
    block: ''
  };
  const toggleStyle = {
    inline: jest.fn(),
    block: jest.fn()
  };

  const controls = shallow(
    <Controls currentStyle={currentStyle}
              toggleStyle={toggleStyle} />
  );
  expect(controls).toMatchSnapshot();
  // expect(controls.find(StyleButton)).toHaveLength(8);
});
