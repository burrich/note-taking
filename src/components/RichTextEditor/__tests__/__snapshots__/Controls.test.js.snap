// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls render correctly 1`] = `
<div
  className="RichEditor-controls"
>
  <InlineStyleButton
    editorStyle={Map {}}
    name="bold"
    onToggle={[Function]}
  />
  <InlineStyleButton
    editorStyle={Map {}}
    name="italic"
    onToggle={[Function]}
  />
  <InlineStyleButton
    editorStyle={Map {}}
    name="underline"
    onToggle={[Function]}
  />
  <InlineStyleButton
    editorStyle={Map {}}
    name="strikethrough"
    onToggle={[Function]}
  />
  <BlockStyleButton
    editorStyle=""
    name="quotes"
    onToggle={[Function]}
  />
  <BlockStyleButton
    editorStyle=""
    name="code"
    onToggle={[Function]}
  />
  <BlockStyleButton
    editorStyle=""
    name="ul"
    onToggle={[Function]}
  />
  <BlockStyleButton
    editorStyle=""
    name="ol"
    onToggle={[Function]}
  />
</div>
`;
